Add ServerNumber variable, sort profile list by server number (default first), fixes for adding/removing profiles, code cleanup.

skittles1 opened the pull request:
- Keep track of server number for each profile.
- Sorts the profiles by server number in the dropdown box, default server number first.
- Displays server number in create account button instead of profile name.
- Add regions to SettingsManager, separate some code in ClientPatchForm.
- Fixes for removing/adding profiles, and switching between modes.

skittles1 committed:
Add ServerNumber field to server profiles.

Hopefully will be able to use this to group profiles in the drop-down
profile list.

skittles1 committed:
Sort profile list by server number, default server first.

skittles1 committed:
Fix removing/adding profiles.

- Add Enabled, SaveProfile and DeleteProfile fields to a new
PatcherSettings created via Add Profile.
- Move Servers list removal code to
SettingsManager.RemoveProfileByName(), called when user removes a
profile using Remove Profile.
- Renamed initial construction of dropdown combobox from RefreshDdl() to
InitDdl().
- Added AddDdl() and RemoveDdl() functions, called when a profile is
added or removed to fix up the dropdown list and select the appropriate
profile.

skittles1 committed:
Code cleanup.

- Added regions to SettingsManager.

Daenks commented on ClientPatcher/SettingsManager.cs:
```diff
@@ -52,52 +59,53 @@ public List<PatcherSettings> GetNewSettings()
+                    // Don't add a server that is supposed to be deleted.
                     if (!webProfile.DeleteProfile)
                         Servers.Add(webProfile);
                 }
```
after the end of this if() statement, we can run another search in local profiles, but instead of matching on Guid, match on the patchinfourl: 
localProfile = Servers.FirstOrDefault(i => i.PatchInfoUrl == webProfile.PatchInfoUrl);
This is how we can locate a profile based on something other than a Guid for updating old profiles (to add a Guid)

skittles1 committed:
Save path from old profiles if set to default.

Users may have two profiles for 103 - an old one with null Guid and a
new one with the correct Guid. If they are using the old one as the
updated profile still (i.e. is set to default), this path needs to be
saved in the new profile (and the new one set to default so it
automatically gets used).
